src/api/transactions.js

```diff
@@ -193,130 +195,242 @@ export async function getTransactions(ctx) {
 
-    // parse childIDs query to get it into the correct format for querying
-    if (filters['childIDs']) {
-      filters['childIDs'] = JSON.parse(filters['childIDs'])
+    ctx.body = {
+      success: true,
+      message: "Tasks created for bulk rerun of transactions" 
     }
+  } catch (e) {
+    console.log(e)
+    utils.logAndSetResponse(
+      ctx,
+      500,
+      `Could not create rerun tasks via the API: ${e}`,
+      'error'
+    )
+  }
+}
 
-    /* Route Filters */
-    // build RegExp for route request path filter
-    if (filters['routes.request.path']) {
-      filters['routes.request.path'] = new RegExp(
-        filters['routes.request.path'],
-        'i'
-      )
-    }
+let transactionsToRerun
 
-    // build RegExp for transaction request querystring filter
-    if (filters['routes.request.querystring']) {
-      filters['routes.request.querystring'] = new RegExp(
-        filters['routes.request.querystring'],
-        'i'
-      )
-    }
+async function createRerunTasks(filters, batchSize, email, page, pages, pauseQueue, filterLimit) {
+  transactionsToRerun = await TransactionModelAPI.find(filters, {_id: 1})
+    .skip(filterLimit*page)
+    .limit(parseInt(filterLimit, 10))
+    .sort({'request.timestamp': -1})
+    .exec()
 
-    // route response status pattern match checking
-    if (
-      filters['routes.response.status'] &&
-      utils.statusCodePatternMatch(filters['routes.response.status'])
-    ) {
-      filters['routes.response.status'] = {
-        $gte: filters['routes.response.status'][0] * 100,
-        $lt: filters['routes.response.status'][0] * 100 + 100
-      }
-    }
+  if (!transactionsToRerun.length) return
 
-    /* orchestration Filters */
-    // build RegExp for orchestration request path filter
-    if (filters['orchestrations.request.path']) {
-      filters['orchestrations.request.path'] = new RegExp(
-        filters['orchestrations.request.path'],
-        'i'
-      )
-    }
+  transactionsToRerun = transactionsToRerun.map(trans => {
+    return {
+      tid: trans._id
+    }})
 
-    // build RegExp for transaction request querystring filter
-    if (filters['orchestrations.request.querystring']) {
-      filters['orchestrations.request.querystring'] = new RegExp(
-        filters['orchestrations.request.querystring'],
-        'i'
-      )
+  const taskObject = {}
+  taskObject.remainingTransactions = transactionsToRerun.length
+  taskObject.user = email
+  taskObject.transactions = transactionsToRerun
+  taskObject.totalTransactions = transactionsToRerun.length
+  taskObject.batchSize = batchSize
+
+  if (pauseQueue) {
+    taskObject.status = 'Paused'
+  }
+
+  const task = await new TaskModelAPI(taskObject).save()
+
+  logger.info(`Rerun task with id ${task._id} created!`)
+
+  if (page < pages) {
+    return createRerunTasks(filters, batchSize, email, ++page, pages, pauseQueue, filterLimit)
```

Comment:
Not sure if we can do this differently, but I think we might eventually run into a large enough dataset for the recursion to use too much memory, probably not an issue now, but just putting it here for future reference. Let me know what you think.

Reply:
We are unlikely to run into issues here. The recursive function does not have local variables. The variables used are global and each time the function is called it replaces the values of those variables and once the recursion is done the variables are set to null